### PR TITLE
Fix for using UglifyJS

### PR DESCRIPTION
**Pull request by ckuijjer**
Solves issue #11

### CODE REVIEW

**Comment by cmswalker** [comment]
Thanks for this, will you try the newest build I just put up on npm?  I wanted to leave the option available for those who dont use UglifyJS.  If your check in this PR was working, it should be good to go.

`npm install fullpage-react@1.0.14`

Also, if you have another PR, you should only have to update the `lib/` directory for PRs and transpile the `dir/` with babel. Thanks again!

**Comment by ckuijjer** [comment]
Hi, thanks for the reply!

I'm afraid `npm install fullpage-react@1.0.14` doesn't seem to solve my server side rendering issue. Looking at [line 212 in fullpage.js](https://github.com/cmswalker/fullpage-react/blob/master/lib/components/fullpage.js#L212) it seems to compare the type to the string 'Slide' instead of to the Slide component itself.

I've forked the [universal-react](https://github.com/ckuijjer/universal-react) boilerplate and added the example from your readme. Starting the client side version using `npm run start` and navigating to `http://localhost:3000/fullpage` shows you can scroll between pages. Starting the version with an initial server side rendering using `npm run start-prod` shows the scrolling unfortunately doesn't work.

If you would like apply my pull request to the example you can change the package.json to use `    "fullpage-react": "ckuijjer/fullpage-react"`

P.S. To be honest I don't completely understand how I should make a PR. Last time I simply changed the files in the `lib/` folder, ran a `npm run build` to transpile the library and bumped the version in `package.json`. What do you prefer the most?

**Comment by ckuijjer** [comment]
Oh, btw I think comparing the type to the Slide component should work both with or without UglifyJS

File: lib/components/fullpage.js [hunk 1]
```diff
@@ -209,7 +210,7 @@ function getSlideCount(children) {
       return result;
     }
 
-    if (c.type.name === 'Slide') {
+    if (c.type === Slide) {
```
Comment (cmswalker) [inline lib/components/fullpage.js]: so this is the only part that doesn't make sense to me. Before being passed through UglifyJS, c.type will equal Function

**Comment by cmswalker** [comment]
I have another idea. What version of babel are you using globally when you run the `npm build` script?    `babel --version`      Noticed that when I run my build and you run yours, they behave differently, which is why my fix from this morning may not have fixed your problem.

**Comment by ckuijjer** [comment]
Awesome! Version 1.0.15 works perfectly! Thanks a lot!

I also didn't know you could do type assertions like that until I looked for a fix to this problem 😀. This seems like a nice way to do it though.

Looking at my package.json I have the following babel packages:

```
"babel-core": "^6.13.2",
"babel-plugin-array-includes": "^2.0.3",
"babel-preset-es2015": "^6.13.2",
"babel-preset-react": "^6.11.1",
"babel-preset-stage-0": "^6.5.0",
"babel-loader": "^6.2.4",
"babel-plugin-react-transform": "^2.0.2",
"babel-plugin-transform-object-assign": "^6.8.0",
```

my `.babelrc` contains

```
{
  "presets": ["es2015", "stage-0", "react"],
  "plugins": ["array-includes"]	
}
```

and when I run `babel --version` it returns `6.10.1 (babel-core 6.9.1)`

**Comment by cmswalker** [comment]
Ah i see ok that makes sense. I'm at `6.11.4 (babel-core 6.13.2)`  so they don't do the __proto__ safe type checking that they used to do. Alright we're good to go! Let me know if you find something else

**Comment by ckuijjer** [comment]
Awesome! Time to update my babel! Thanks again for adding this to your library!

**Pull request closed without merging**